File: 70-climbing-stairs/70-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if (n == 1){
      return 1
    }
    let dp = [n+1, 1, 2]
    for (let i = 3; i <= n; i++){
      dp[i] = dp[i-1] + dp[i-2];
    }
    return dp[n]
};

File: 70-climbing-stairs/70-climbing-stairs.cpp
#include <vector>
class Solution {
public:
  int climbStairs(int n) {
    if (n == 1) {
      return 1;
    }
    std::vector<int> dp = {(n + 1), 1, 2};
    for (int i = 3; i <= n; ++i) {
       dp.push_back(dp[i - 1] + dp[i - 2]);
    }
    return dp[n];
  }
};
